mod-invoice-storage/schemas/voucher.json

```diff
@@ -0,0 +1,117 @@
+    },
+    "exchangeRate": {
+      "description": "Exchange rate used to generate transaction",
+      "type": "number"
+    },
+    "exportToAccounting": {
+      "description": "Export this payment record to a financial system",
+      "type": "boolean"
+    },
+    "voucherType":{
+      "description": "Indicates type of payment to account",
+      "type": "string",
+      "enum": [
+        "Payment",
+        "Pre-Payment",
+        "Credit",
+        "Voucher"
+      ]
+    },
+    "voucherStatus":{
+      "description": "status of the voucher",
+      "type": "string",
+      "enum": [
+        "Awaiting Payment",
+        "Paid"
+      ]
+    },
+    "voucherAmount": {
+      "description": "The amount that is actually paid",
+      "type": "number"
+    },
+    "voucherDate": {
+      "description": "The date the invoice status changes to approved",
+      "type": "string",
+      "format": "date-time"
+    },
+    "voucherNumber": {
+      "description": "Invoice number in folio system",
+      "type": "string"
+    },
+    "batchNumber": {
+      "description": "Only required when voucher is being output automatically",
+      "type": "string"
+    },
+    "invoiceId": {
+      "description": "Corresponding invoce Id",
+      "type": "string",
+      "pattern": "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$"
+    },
+    "userId": {
+      "description": "ID of the user who transitioned the invoice to approved",
+      "type": "string",
+      "pattern": "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$"
+    },
+    "voucherLines": {
+      "description": "references the voucher lines associated to the voucher",
+      "id": "voucherLines",
+      "type": "array",
+      "items": {
+        "type": "object",
+        "$ref": "voucher_line.json"
+      }
+    },
+    "accountingCode": {
+      "description": "Number that represents the vendor is an external accounting system;This is the number as stated on the invoice",
+      "type": "string"
+    },
+    "disbursementNumber": {
+      "description": "The identifier for the physical transaction corresponding to a payment;Brought in from external source",
+      "type": "string"
+    },
+    "disbursementDate":{
+      "description": "Date payment was made from financia system",
+      "type": "string",
+      "format": "date-time"
+    },
+    "disbursementAmount":{
+      "description": "The amount of the payment made outside of FOLIO",
+      "type": "number"
+    },
+    "metadata": {
+      "type": "object",
+      "$ref": "../../../raml-util/schemas/metadata.schema",
+      "readonly": true
+    }
+  },
+  "additionalProperties": false,
+  "required": [
+     "systemCurrency",
+     "invoiceCurrency",
+     "exportToAccounting",
+     "voucherType",
+     "voucherStatus",
+     "voucherAmount",
+     "voucherNumber",
+     "batchNumber",
+     "invoiceId",
+     "accountingCode"
```

Comment:
As per the interface documentation `userId` is a required field. Should we add it in the required section here?

Reply:
The notes section mentions "**Potentially need to capture a user id when possible"** . Possibly a question for @craigmcnally

Reply:
I would say for now just remove the userId field altogether and rely on the metadata createdBy, updatedBy, etc.

Reply:
removed the userId field